scripts/drs_file_access_examination.ipynb

```diff
@@ -0,0 +1,326 @@
+    "for table in data_tables_list:\n",
+    "    table_colums = table[\"attributes\"]\n",
+    "    object_id_column = dict(filter(lambda item: column_name_search_key in item[0], table_colums.items())) \n",
+    "    \n",
+    "    # Checking if the object_id column's contents is a string and start with 'drs://' \n",
+    "    if object_id_column:\n",
+    "        DRS_url = next(iter(object_id_column.values()))\n",
+    "        if type(DRS_url) is str and DRS_url[0:6] == 'drs://':\n",
+    "            DRS_urls_dict[DRS_url] = {'table_name': table['entityType'], \n",
+    "                                       'row_id' : table['name'], \n",
+    "                                       'drs_url' : DRS_url}\n",
+    "            \n",
+    "# # Uncomment this code if you want to run the notebook on the first 10 DRS url in the data table\n",
+    "# for i in range(len(DRS_urls_dict)-10):\n",
+    "#     DRS_urls_dict.popitem()\n",
+    "    \n",
+    "# Outputting the amount of DRS urls found\n",
+    "print(\"There was \" + str(len(DRS_urls_dict)) + \" DRS found in this workspace\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Resolve DRS URIs and Checking File Access\n",
+    "\n",
+    "DRS creates a unique ID mapping that allows for flexible file retrieval. The unique mapping is the DRS Uniform Resource Identifier (URI) - a string of characters that uniquely identifies a particular cloud-based resource (similar to URLs) and is agnostic to the cloud infrastructure where it physically exists. To learn where the file physically exists on the cloud, we must resolve the DRS through a backend service called Martha that will unmap the DRS url to get the file's google bucket file path.\n",
+    "\n",
+    "In this step, we check if all the DRS urls can be resolved and check our access to the file it points to in the google bucket. If the DRS url can't be resolved, we will record the error that will, later on, be shown in the report in the last step.\n",
+    "\n",
+    "One of the most comment reasons the DRS urls can not resolve is \"Fence is not linked\". If you have this error, it probably because the data is controlled-access. To use controlled-access data on Terra, you will need to link your Terra user ID to your authorization account (such as a dbGaP account). Linking to external servers will allow Terra to automatically determine if you can access controlled datasets hosted in Terra (ex. TCGA, TOPMed, etc.) based on your valid dbGaP applications. Go to this link to learn more more: https://support.terra.bio/hc/en-us/articles/360038086332\n",
+    "\n",
+    "## Estimated time \n",
+    "\n",
+    "This code resolved and check file access to 50 DRS url a 1 min\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
```

Comment:
i find it generally nice to put all your imports at the top of the notebook - this may be personal preference but it's frustrating to me to get to the middle of a notebook and only then realize i don't have a package installed and have to re-run everything after installing the package